sortedm2m/fields.py

```diff
@@ -376,17 +382,19 @@ def get_intermediate_model_to_field(self, klass):
 
     def create_intermediate_model_from_attrs(self, klass, attrs):
         name = self.get_intermediate_model_name(klass)
-        return type(str(name), (models.Model,), attrs)
+        if self.base_class:
+            return type(str(name), (models.Model, self.base_class), attrs)
+        else:
+            return type(str(name), (models.Model,), attrs)
```

Comment:
this whole if block can be improved. You are repeating `str(name)`, `(models.Model,)` and `attrs` twice.

You could store that tuple in an auxiliary variable and insert the base class if needed.

Could you try this? You could also try a different idea if you like.